### DISCUSSION TITLE
Accessing settings in Blade

### DISCUSSION THREAD

**Discussion by nwconsulting** [spatie/laravel-settings#98] [Q&A]
Sorry for the nubee question.

I am trying to access the settings in Blade but not sure how to do this. I want to be able check if a setting has a specific value and if so then do something.

**Comment by gboquizosanchez** [comment]
Hi @nwconsulting you can create an instance through app(GeneralSettings::class) directly in blade with the whole namespace of the class, for example:

`$settings = app(\App\Settings\GeneralSettings::class);`

Or

`$settings = new \App\Settings\GeneralSettings();`

But ideal I think is to use a controller to give the responsibility of data to it and pass it to the view.

I hope this can be helpfully for you.

Regards.

**Comment by Neography7** [comment]
Actually, I created a helper class for calling settings in Blade

```
<?php

namespace App\Utilities;

use App\Settings\GeneralSettings;

class CallSettings {

  public function get ($settings_key)
  {
  
      $settings = new GeneralSettings;
      return $settings->$settings_key;
  
  }
}
```

And I added this class to Config/App.php under aliases

`'CallSettings' => App\Utilities\CallSettings::class,`

Now I use this for calling settings in blade pages

`CallSettings("site_name")`

I don't know whether as it's supposed to be or not but it works as I want for me now :)

**Comment by merlinro** [comment]
I have passed the settings like this in the `AppServiceProvider `

```
namespace App\Providers;

use App\Settings\GeneralSettings;
use Illuminate\Support\Facades\View;
use Illuminate\Support\ServiceProvider;

class AppServiceProvider extends ServiceProvider
{
    /**
     * Register any application services.
     *
     * @return void
     */
    public function register()
    {
    }

    /**
     * Bootstrap any application services.
     *
     * @return void
     */
    public function boot()
    {
        View::share( 'settings',new \App\Settings\GeneralSettings());
    }
}
```
